CONTRIBUTING.md

```diff
@@ -0,0 +1,900 @@
+
+### Line length
+
+Keep lines down to a maximum of 80 characters. Minifiers are very good at compressing whitespace, so don't be afraid to use it.
+
+The only exception to this rule is with long regexes. Append `// eslint-disable-line max-len` to the end of those lines.
+
+### Function length
+
+Keep function length to no more than 20 statements, including those in nested blocks.
+
+The only exceptions are for revealing module wrappers and Mocha `describe` and `context` blocks.
+
+This isn't checked for the tests, but still, keep it reasonable.
+
+### Line endings
+
+Use Windows-style line endings (i.e. CRLF).
+
+### Semicolons
+
+Avoid semicolons. (A few resources to help you understand this: [*](http://blog.izs.me/post/2353458699/an-open-letter-to-javascript-leaders-regarding) [*](http://inimino.org/~inimino/blog/javascript_semicolons) [*](http://jamesallardice.com/understanding-automatic-semi-colon-insertion-in-javascript/))
+
+### Redundant expressions
+
+Don't use redundant parentheses, curly braces, `undefined`s, labels, ternaries, etc. Examples:
+
+- `var foo = value === 1 ? true : false` is better written as `var foo = value === 1`.
+- `var foo = (value === 1) ? bar : baz` is better written as `var foo = value === 1 ? bar : baz`.
+- `var foo = undefined` is better written as `var foo`.
+- `return undefined` is better written as `return`.
+- `delete(obj.foo)` is better written as `delete obj.foo`.
+- `return` at the end of a function if you're returning `undefined` should just be omitted.
+- If you're returning `null`, there's usually little reason to not just return `undefined`.
+- Unused variables should be removed.
+- An exception is detailed below for assignment in conditions.
+
+### Equality testing
+
+Use strict equality (`===`/`!==`), not loose equality (`==`/`!=`). The only exception is if you're testing for `null` or `undefined`, in which loose equality, `value == null` and `value != null` is preferred. The reason is for type safety with primitives.
+
+### `eval` and friends
+
+Never use `eval`, the `Function` constructor, and friends. Those will fail when running under CSP restrictions, and this library should work in those environments.
+
+### Quotes
+
+Prefer double quotes to single quotes, but using single quotes to avoid escaping is okay.
+
+### Strict mode
+
+All code must be in strict mode, preferably wrapped in an IIFE.
```

Comment:
This is not strictly true. Code running at the outermost context should not be in strict mode because it can break concat scripts